### PR TITLE
Update helio

### PR DESCRIPTION
**Pull request by romange**
Add support for asan/ubsan checkers in our dev environment.
Once we fix all the problems we will enable them in our CI as well.
Fix most of the clang warnings

### CODE REVIEW

File: src/server/CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,9 @@
+option(DF_ENABLE_MEMORY_TRACKING "Adds memory tracking debugging via MEMORY TRACK command" ON)
```
Comment (romange) [inline src/server/CMakeLists.txt:1]: just reshuffling code here

File: src/server/zset_family.cc [hunk 2]
```diff
@@ -3268,7 +3268,7 @@ void ZSetFamily::Register(CommandRegistry* registry) {
       << CI{"ZREM", CO::FAST | CO::WRITE, -3, 1, 1, acl::kZRem}.HFUNC(ZRem)
       << CI{"ZRANGE", CO::READONLY, -4, 1, 1, acl::kZRange}.HFUNC(ZRange)
       << CI{"ZRANDMEMBER", CO::READONLY, -2, 1, 1, acl::kZRandMember}.HFUNC(ZRandMember)
-      << CI{"ZRANK", CO::READONLY | CO::FAST, 3, 1, 1, acl::kZRange}.HFUNC(ZRank)
+      << CI{"ZRANK", CO::READONLY | CO::FAST, 3, 1, 1, acl::kZRank}.HFUNC(ZRank)
```
Comment (kostasrim) [inline src/server/zset_family.cc:3271]: nice catch!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -94,9 +106,15 @@ jobs:
       - name: Build
         run: |
           cd ${GITHUB_WORKSPACE}/build
+          ninja search_family_test
```
Comment (kostasrim) [inline .github/workflows/ci.yml:109]: Why we need this?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -53,20 +53,32 @@ jobs:
...
         with:
           submodules: true
-      - name: Install dependencies
+      - name: Prepare Environment
         run: |
           uname -a
           cmake --version
           mkdir -p ${GITHUB_WORKSPACE}/build
 
+          echo "===================Before freeing up space ============================================"
```
Comment (kostasrim) [inline .github/workflows/ci.yml:73]: Why is this needed?

**Review by kostasrim — COMMENTED** [review]
I am especially happy for the unqualified -> qualified calls to `move` (We had a lot of warnings from these)

**Review by BorysTheDev — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (romange) [reply .github/workflows/ci.yml:73]: because `-fsanitize` eats up all the space on the runner and we fail building the repo

File: .github/workflows/ci.yml (hunk 3 above)
Comment (romange) [reply .github/workflows/ci.yml:109]: it's an arbitrary target that we would build anyway provides a consistent target being built. helps if there is a failure like disk space which gets lost when we build multiple targets in parallel.
